sdk/tables/Azure.Data.Tables/src/Sas/TableAccountSasBuilder.cs

```diff
@@ -0,0 +1,233 @@
+        /// <param name="resourceTypes"><see cref="TableAccountSasResourceTypes"/> containing the accessible resource types.</param>
+        /// <param name="expiresOn">The time at which the shared access signature becomes invalid.</param>
+        public TableAccountSasBuilder(string rawPermissions, TableAccountSasResourceTypes resourceTypes, DateTimeOffset expiresOn)
+        {
+            ExpiresOn = expiresOn;
+            Permissions = rawPermissions;
+            ResourceTypes = resourceTypes;
+        }
+
+        /// <summary>
+        /// The optional signed protocol field specifies the protocol
+        /// permitted for a request made with the SAS.  Possible values are
+        /// <see cref="SasProtocol.HttpsAndHttp"/>,
+        /// <see cref="SasProtocol.Https"/>, and
+        /// <see cref="SasProtocol.None"/>.
+        /// </summary>
+        public SasProtocol Protocol { get; set; }
+
+        /// <summary>
+        /// Optionally specify the time at which the shared access signature
+        /// becomes valid.  If omitted when DateTimeOffset.MinValue is used,
+        /// start time for this call is assumed to be the time when the
+        /// storage service receives the request.
+        /// </summary>
+        public DateTimeOffset StartsOn { get; set; }
+
+        /// <summary>
+        /// The time at which the shared access signature becomes invalid.
+        /// This field must be omitted if it has been specified in an
+        /// associated stored access policy.
+        /// </summary>
+        public DateTimeOffset ExpiresOn { get; set; }
+
+        /// <summary>
+        /// The permissions associated with the shared access signature. The
+        /// user is restricted to operations allowed by the permissions. This
+        /// field must be omitted if it has been specified in an associated
+        /// stored access policy.  <see cref="TableAccountSasPermissions"/> can be used to create the
+        /// permissions string.
+        /// </summary>
+        public string Permissions { get; private set; }
+
+        /// <summary>
+        /// Specifies an IP address or a range of IP addresses from which to
+        /// accept requests. If the IP address from which the request
+        /// originates does not match the IP address or address range
+        /// specified on the SAS token, the request is not authenticated.
+        /// When specifying a range of IP addresses, note that the range is
+        /// inclusive.
+        /// </summary>
+        public SasIPRange IPRange { get; set; }
+
+        /// <summary>
+        /// An optional unique value up to 64 characters in length that
+        /// correlates to an access policy specified for the container.
+        /// </summary>
+        public string Identifier { get; set; }
```

Comment:
Is this a valid field for account level SAS?

Reply:
I got this off of the Storage.Common `SasQueryParameters`, but you're right that it's not valid for account level. I think this will be resolved when Ted weighs in on how the Query Parameters class hierarchy should be.